pkg/controllers/job/plugins/distributed-framework/mpi/mpi_test.go

```diff
@@ -61,26 +67,47 @@ func TestMpi(t *testing.T) {
 				Spec: v1alpha1.JobSpec{
 					Tasks: []v1alpha1.TaskSpec{
 						{
-							Name:     "fakeMaster",
+							Name:     "master",
 							Replicas: 1,
 							Template: v1.PodTemplateSpec{},
 						},
 						{
-							Name:     "fakeWorker",
+							Name:     "worker",
 							Replicas: 2,
 							Template: v1.PodTemplateSpec{},
 						},
 					},
 				},
+				Status: v1alpha1.JobStatus{
+					ControlledResources: map[string]string{},
+				},
 			},
 			Pod: &v1.Pod{
```

Comment:
How about use BuildPod？

Reply:
Indeed, using BuildPod can reduce redundant code

Reply:
I took a look at the source code of BuildPod, and it may not be enough to set some fields, such as the InitContainers field. I don’t want to change other code that has nothing to do with this PR, so I think it would be better to keep it as it is.

Reply:
@Monokaix WDYT